packages/vue3-vite/src/external/copy-package-json-plugin.ts

```diff
@@ -0,0 +1,21 @@
+import fs from 'fs';
+import path from 'path';
+import { OutputPlugin } from 'rollup';
+
+/**
+ * Copy `package.json` to `dist` folder if it exists.
+ * @param appDir Directory path where `package.json` is located.
+ */
+export const copyPackageJsonPlugin = (appDir: string): OutputPlugin => ({
```

Comment:
I add this plugin for `build.rollupOptions.plugins` so that we can copy `package.json` to the desired `dist` folder during build.
I use [outputOptions](https://rollupjs.org/plugin-development/#outputoptions) instead of [build hooks](https://rollupjs.org/plugin-development/#build-hooks) because this is the only place I found that allows us to retrieve the output directory.